File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {    
    const converted = n.toString(k).split("0")
    return converted.reduce((ans, val) => ans += is_prime(Number(val)) ? 1 : 0, 0);        
}

function is_prime(num) {
    if (num < 2) return false
    for (let i = 2; i * i <= num; i++) {
        if (num % i == 0) return false
    }
    return true
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        
        String[] converted_num = Integer.toString(n, k).split("0");
        
        for (String num : converted_num) {
            if (num.isEmpty()) {
                continue;
            }            
            answer += is_prime_number(Long.parseLong(num)) ? 1 : 0;
        }
        return answer;
    }
    
    private boolean is_prime_number(long num) {
        if (num < 2) {
            return false;
        }
        for (long i = 2; i * i <= num; i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}